src/node_sqlite.cc

```diff
@@ -506,6 +508,139 @@ void StatementSync::All(const FunctionCallbackInfo<Value>& args) {
+  auto isolate = env->isolate();
+
+  Local<External> data = Local<External>::Cast(args.Data());
+  IterateCaptureContext* capture_context =
+      static_cast<IterateCaptureContext*>(data->Value());
+  auto stmt = capture_context->stmt;
+  auto num_cols = capture_context->num_cols;
+
+  int r = sqlite3_step(stmt->statement_);
+  if (r != SQLITE_ROW) {
+    CHECK_ERROR_OR_THROW(
+        env->isolate(), stmt->db_->Connection(), r, SQLITE_DONE, void());
+    sqlite3_reset(stmt->statement_);
+
+    LocalVector<Name> keys(isolate, {env->done_string(), env->value_string()});
+    LocalVector<Value> values(isolate,
+                              {Boolean::New(isolate, true), Null(isolate)});
+
+    Local<Object> result =
+        Object::New(isolate, Null(isolate), keys.data(), values.data(), 2);
+    args.GetReturnValue().Set(result);
+    return;
+  }
+
+  LocalVector<Name> row_keys(isolate);
+  row_keys.reserve(num_cols);
+  LocalVector<Value> row_values(isolate);
+  row_values.reserve(num_cols);
+  for (int i = 0; i < num_cols; ++i) {
+    Local<Name> key;
+    if (!stmt->ColumnNameToName(i).ToLocal(&key)) return;
+    Local<Value> val;
+    if (!stmt->ColumnToValue(i).ToLocal(&val)) return;
+    row_keys.emplace_back(key);
+    row_values.emplace_back(val);
+  }
+
+  Local<Object> row = Object::New(
+      isolate, Null(isolate), row_keys.data(), row_values.data(), num_cols);
+
+  LocalVector<Name> keys(isolate, {env->done_string(), env->value_string()});
+  LocalVector<Value> values(isolate, {Boolean::New(isolate, false), row});
+
+  Local<Object> result =
+      Object::New(isolate, Null(isolate), keys.data(), values.data(), 2);
+  args.GetReturnValue().Set(result);
+}
+
+void StatementSync::Iterate(const FunctionCallbackInfo<Value>& args) {
+  StatementSync* stmt;
+  ASSIGN_OR_RETURN_UNWRAP(&stmt, args.This());
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_ON_BAD_STATE(
+      env, stmt->IsFinalized(), "statement has been finalized");
+  auto isolate = env->isolate();
+  auto context = env->context();
+  int r = sqlite3_reset(stmt->statement_);
+  CHECK_ERROR_OR_THROW(
+      env->isolate(), stmt->db_->Connection(), r, SQLITE_OK, void());
+
+  if (!stmt->BindParams(args)) {
+    return;
+  }
+
+  IterateCaptureContext* capture_context = new IterateCaptureContext();
+  capture_context->num_cols = sqlite3_column_count(stmt->statement_);
+  capture_context->stmt = stmt;
+
+  Local<Function> next_func =
+      Function::New(context,
+                    StatementSync::IterateNextCallback,
+                    External::New(isolate, capture_context))
+          .ToLocalChecked();
+  Local<Function> return_func =
+      Function::New(context,
+                    StatementSync::IterateReturnCallback,
+                    External::New(isolate, capture_context))
+          .ToLocalChecked();
+
+  LocalVector<Name> keys(
+      isolate,
+      {env->next_string(), String::NewFromUtf8Literal(isolate, "return")});
```

Comment:
you can add `return_string` to PER_ISOLATE_STRING_PROPERTIES (in `env_properties.h`) and use it here.

Reply:
Done